sunpy/data/manager/cache.py

```diff
@@ -0,0 +1,93 @@
+from urllib.request import urlopen
+
+from sunpy.util.util import hash_file, replacement_filename
+from sunpy.util.net import get_filename
+
+
+class Cache:
+    """
+    Cache handles caching.
+    """
+
+    def __init__(self, downloader, storage, cache_dir):
+        self._downloader = downloader
+        self._storage = storage
+        self._cache_dir = cache_dir
+        if not self._cache_dir.endswith('/'):
+            self._cache_dir += '/'
```

Comment:
I would suggest to use `Pathlib`, and direcly append it to whatever after with `_cache_dir / file`